`params`: support for tracking all params in a file (#3431)
iterative/dvc.org: content/docs/command-reference/params/index.md

changes to other parts of the params file will not affect the stage. Parameter
dependencies also prevent situations where several stages share a regular
dependency (e.g. a config file), and any change in it invalidates all of them
(see `dvc status`), causing unnecessary re-executions upon `dvc repro`. However,
you can track all the parameters in a file by setting _parameter value_ to be

> `null`.

Author: I am not quite sure on how to include this information here.

Reviewer: I think an example either here or in the examples section below would clarify better than any explanation.

Reviewer: It would help to show both `dvc.yaml` and how `dvc stage add` can be used.

Some rationale for it might also help, like explaining that you may want to granularly track all parameters in a file and it would be tedious to list them all and keep them updated as new parameters are added.

Reviewer 2: No need to include this info. in this paragraph I think. This is a general explanation about parameter dependencies.

How about adding the note or even a new paragraph and sample block to https://dvc.org/doc/user-guide/project-structure/pipelines-files#parameter-dependencies instead?

Agree about updating `stage add` (as I mentioned in another comment) and on mentioning motivation when presenting this function

Reviewer 2: Will update the other refs as agreed. Deleting this for now...
```suggestion
(see `dvc status`), causing unnecessary re-executions upon `dvc repro`.
```

> you can track all the parameters in a file by setting _parameter value_ to be
> `null`.

Reviewer: Yeah, this is tricky. What about this?

```suggestion
you can track all the parameters in a file by specifying the value of
`params.yaml:` to be `null` or empty.
```

Reviewer 2: I modified the sample instead (similar to the `dvc.yaml` ref.) and generalized the text in the description. See 0ef8c607
